presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/PrecisionRecallStateSerializer.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.operator.aggregation.fixedhistogram.FixedDoubleHistogram;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+import io.airlift.slice.SizeOf;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+
+import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class PrecisionRecallStateSerializer
+        implements AccumulatorStateSerializer<PrecisionRecallState>
+{
+    @Override
+    public Type getSerializedType()
+    {
+        return VARBINARY;
+    }
+
+    @Override
+    public void serialize(PrecisionRecallState state, BlockBuilder out)
+    {
+        int requiredBytes = SizeOf.SIZE_OF_BYTE; // has histograms;
+        if (state.getTrueWeights() != null) {
+            requiredBytes += state.getTrueWeights().getRequiredBytesForSerialization();
+            requiredBytes += state.getFalseWeights().getRequiredBytesForSerialization();
+        }
+
+        SliceOutput sliceOut = Slices.allocate(requiredBytes).getOutput();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *sliceOut*. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).